src/main/java/com/example/provider/VehicleServlet.java

```diff
@@ -82,35 +88,47 @@ public VehicleServlet(
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     ServletUtils.setStandardResponseHeaders(response);
-
+    
     String vehicleId;
 
-    try {
-      vehicleId = ServletUtils.getEntityIdFromRequestPath(request);
-    } catch (IllegalArgumentException e) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "No Vehicle id provided.");
-      return;
-    }
+    vehicleId = ServletUtils.getEntityIdFromRequestPath(request);
 
-    logger.info(String.format("Getting vehicle with vehicleID: %s", vehicleId));
+    if (vehicleId != null && !vehicleId.isEmpty()) {
+      logger.info(String.format("Getting vehicle with vehicleID: %s", vehicleId));
 
-    GetVehicleRequest getVehicleRequest =
+      GetVehicleRequest getVehicleRequest =
         GetVehicleRequest.newBuilder().setName(VehicleUtils.getVehicleName(vehicleId)).build();
 
-    Vehicle vehicle = vehicleServiceClient.getVehicle(getVehicleRequest);
-    servletState.setVehicleId(vehicleId);
+      Vehicle vehicle = vehicleServiceClient.getVehicle(getVehicleRequest);
+      servletState.setVehicleId(vehicleId);
 
-    logger.info(String.format("Vehicle:\n%s", vehicle));
+      logger.info(String.format("Vehicle:\n%s", vehicle));
 
-    if (vehicle != null) {
-      if (tripMatcher.triggerMatching(vehicle, vehicleId)) {
-        vehicle = vehicleServiceClient.getVehicle(getVehicleRequest);
+      if (vehicle != null) {
+        if (tripMatcher.triggerMatching(vehicle, vehicleId)) {
+          vehicle = vehicleServiceClient.getVehicle(getVehicleRequest);
+        }
       }
-    }
 
-    response.getWriter().print(GsonProvider.get().toJson(vehicle));
-    response.getWriter().flush();
+      response.getWriter().print(GsonProvider.get().toJson(vehicle));
+      response.getWriter().flush();
```

Comment:
Nit: Instead of having the `if-else`, I'd just keep one `if` and do an early return. 
The main benefit of this is in the case that some day we add code below line 129 (`response.getWriter().flush();` doesn't exit the method)

Reply:
Done.